CONTRIBUTING.md

```diff
@@ -19,8 +19,57 @@ We heartily welcome pull requests. To ensure an effective contribution, please a
 1. For significant, invasive, or output-affecting changes, consider opening an issue for discussion before committing substantial time and effort.
 2. If you're new to the project, starting with the [Good first issues](https://github.com/ocaml-ppx/ocamlformat/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+label%3A%22Good-first-issue+%3Agreen_heart%3A%22) can be beneficial.
 3. Fork the repository and create your branch from `main`.
-4. If you've added code that should be tested, supplement it with tests located in the `test/` directory.
-5. Ensure that the test suite passes (see [Running the tests](#running-the-tests) for instructions).
+4. Ensure that the test suite passes (see [Running the tests](#running-the-tests) for instructions). If there are changes in the tests,
+   verify that they are expected. If no existing test output changed, you likely need to add new tests (see [Adding a test](#adding-a-test)).
+
+### Adding a Test
+
+#### Unit Tests
+
+Unit tests are located in the `test/unit` directory and are written using [Alcotest](https://github.com/mirage/alcotest).
+
+#### Golden Tests
+
+The majority of tests for `ocamlformat` are so called "golden" or "expect" tests. These
+tests all consist of running `ocamlformat` on `.ml` files and comparing the
+output to expected output stored in reference files. The `dune` configuration
+for this is automatically generated for you based on the files present in these
+directories.
+
+These are separated into `test/failing` and `test/passing` directories.
+
+To add a test showing currently incorrect behavior, add a `.ml` file to
+`test/failing/tests`. If command line arguments are needed, create a
+corresponding `.opts` file with the same base name. The output of these tests
```

Comment:
Base name is ambiguous here, I think you should use entire file names. For example, it's not `mytest.opts` but `mytest.ml.opts`